src/lib/colorUtils.ts

```diff
@@ -9,3 +17,43 @@ export const mapColorsToScheme = <T>(
     { ...siteVars.emphasisColors, ...siteVars.naturalColors },
     typeof mapper === 'number' ? String(mapper) : (mapper as any),
   ) as ColorValues<T>
+
+export const getColorSchemeFn = <T>(colorProp: string, colorScheme: ColorValues<T>) => {
+  const colors = _.get(colorScheme, colorProp)
+  return (area: keyof T, defaultColor: string) => (colors ? colors[area] : defaultColor)
+}
+
+export const getColorSchemeFromObject = (
+  colorScheme: ColorValues<Partial<ColorScheme>>,
+  colors: ComplexColorPropType,
+): Partial<ColorScheme> =>
+  _.mapValues(colors, (color, colorName) => {
+    // if the color scheme contains the color, then get the value from it, otherwise return the color provided
```

Comment:
This idea looks quite confusing to me :(

I have a definition for `blue` in the color scheme, but `blue` is a valid CSS color, https://www.quackit.com/css/css_color_codes.cfm

Reply:
Totally understand your point, but I don't have better proposal at this moment. We have to allow the user somehow to define the colors by name (the ones we have in the color palette, but still if we don't have them, we would end up with the CSS color.. If they define it as hex value then it is clear, but in the other case, is kind a confusing. Still, we have to use color names for the color prop, so I don't have any other proposal at this moment)

Reply:
```jsx
// uses `red` from schema
<Label color='red' />

// uses CSS `red`, i.e. #FF0000
<Label color={{ background: 'red' }} />

// still uses CSS colors, but you can get them from schema
<Label color={(colorSchema) => ({ background: colorSchema.red.background })} />
```

What about `color` as function?

Reply:
Couple of thoughts on this: 
- not sure what would be the use cases where the user will want to use the CSS colors, when they have defined schema for which colors will use
- more-over why should the prop in our API as **default** returns just some CSS color - as a user would make me wonder why we have special API for that

On the other hand, I understand the confusion, but would rather not over-complicate the API now, and wait for complains if there will be any in the future.

Reply:
totally support @mnajdova 's points

if the color scheme name collides with the CSS color name it's by design and it's expected since the developer should stick to a theme color as much as possible

if the developer reaaaally wants the actual CSS red (which is not recommended since they're supposed to work within the theme constraints), they can still use the hexcode (#FF0000)